File: Data Structures & Algorithms/binary-search/submission-0.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int start = 0, end=nums.size()-1;
        while(start<=end)
        {
            int middle = end+(start-end) /2;
            if(nums[middle]==target)
            {
                return middle;
            }
            if(nums[middle]<target)
            {
                start = middle+1;
            }
            else
            {
                end = middle-1;
            }
        }
        return -1;
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        myMap = {}
        for i in range(len(nums)):
            if target-nums[i] in myMap:
                return [myMap[target-nums[i]],i]
            myMap[nums[i]] = i;
